src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -21,5 +21,11 @@ public interface IWorkspaceModel
         event Action NodesCleared;
         event Action<ConnectorModel> ConnectorAdded;
         event Action<ConnectorModel> ConnectorDeleted;
+
+        /// <summary>
+        /// Implement to record node modification for undo/redo
+        /// </summary>
+        /// <param name="nodeModel"></param>
+        void RecordModelForModification(ModelBase nodeModel);
```

Comment:
@pboyer you're right, the `CommandExecutive`'s `ExecuteCommand` will handle recording for undo. The `RecordModelForModification` API needs to be exposed to record a manipulator "move" action for undo. It needs to be called on the `MouseDown` event of the manipulator so that it can start recording any of its actions from that point on.